content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Long/1. Long.md

```diff
@@ -0,0 +1,55 @@
+---
+title: 1. Long
+taxonomie:
+  - ib-19.2.Datatype-Primitief-Long.OI
+  - ib-19.3.Datatype-Primitief-Long.OI
+  - ib-19.2.Datatype-Primitief-Ulong.OI
+  - ib-19.3.Datatype-Primitief-Ulong.OI
+---
+
+## Wat is een Long?
+Het `long` datatype in C# is een numeriek datatype dat bedoeld is voor het opslaan van grotere gehele getallen dan een `int` kan bevatten. Een `long` kan waarden opslaan tussen -9,223,372,036,854,775,808 en 9,223,372,036,854,775,807 en gebruikt 64 bits voor opslag.
+
+> [!TIP] Casus: Stel je wilt de wereldbevolking opslaan in je programma, omdat je hiermee verdere berekeningen gaat uitvoeren. Hiervoor heb je een datatype nodig dat grote getallen aankan.
+
+## Hoe zit een Long in elkaar?
+Een `long` is een datatype dat grotere hele getallen kan opslaan doordat het meer bits gebruikt, namelijk 64 bits. Het heeft een bereik van extreem grote negatieve tot grote positieve getallen, wat het geschikt maakt voor situaties waarin je getallen nodig hebt die buiten het bereik van een `int` vallen.
+
+### Codevoorbeeld
+```C#
+long population = 7800000000; // Wereldbevolking in een long variabele
+Console.WriteLine("De huidige wereldbevolking is: " + population);
+```
+
+In dit voorbeeld wordt de wereldbevolking opgeslagen in een `long` variabele. De waarde is veel te groot voor een `int`, maar kan met `long` gemakkelijk worden opgeslagen.
+
+## Wanneer gebruik je een Long?
+Een `long` gebruik je wanneer je gehele getallen wilt opslaan die groter zijn dan het bereik van een `int`. Dit datatype is handig in situaties waar je grote getallen moet verwerken, zoals bij wetenschappelijke berekeningen, grote datasets of financiële toepassingen. Omdat een `long` echter alleen gehele getallen kan opslaan, is het niet geschikt voor decimalen of nauwkeurige berekeningen. Voor werk met grote getallen die ook decimalen bevatten, zijn `double` of `decimal` geschiktere keuzes.
+
+> [!TIP] Casus: Stel dat je de totale jaarlijkse verkopen van een groot bedrijf bijhoudt. Je hebt hiervoor mogelijk getallen in de miljarden nodig, die buiten het bereik van een `int` vallen.
+
+**Mogelijke uitwerking van de casus**
+```C#
+long jaarlijkseVerkopen = 8500000000; // Totale jaarlijkse verkoop in een long variabele
+Console.WriteLine("Totale jaarlijkse verkopen zijn: " + jaarlijkseVerkopen + " euro");
+```
+
+Dit voorbeeld laat zien hoe je grote getallen, zoals jaarlijkse verkopen in de miljarden, kunt opslaan met een `long`.
+
+> [!Warning] Let op: Wanneer een `long`-waarde het maximum bereik overschrijdt, kan dit leiden tot een overflow, waarbij het getal bij het laagste getal opnieuw begint. Dit kan worden voorkomen door het gebruik van `checked` om overflow-fouten op te sporen.
+
+## Wat is een ULong?
```

Comment:
Overlegd met Teun en dit kan toch beter naar een tweede, los bestand. Om dan werk te besparen vinden we het prima als je voor Wat is het en Hoe zit het in elkaar verwijst naar Long. Dus wel die kopjes, maar dan een zin waarin je verwijst naar long. 

PI kan je overslaan.
DT kan je wel gewoon splitsen, maar dan misschien een verwijzing van long naar ulong en andersom.

Het is prima alles in dezeflde map te houden. Dus `OI/.../long/1. Long.md` `OI/.../long/1. ULong.md` etc